tests/test_report.py

```diff
@@ -0,0 +1,68 @@
+import json
+import subprocess
+import sys
+from pathlib import Path
+
+import pytest
+
+from unblob.models import ProcessResult, TaskResult
+from unblob.report import ExtractCommandFailedReport
+
+
+@pytest.fixture
+def input_file(tmp_path: Path) -> Path:
+    input_file = tmp_path / "input_file"
+    input_file.write_bytes(bytes(range(256)))
+    return input_file
+
+
+@pytest.fixture
+def extract_root(tmp_path: Path) -> Path:
+    return tmp_path / "extract_root"
+
+
+@pytest.fixture
+def report_file(tmp_path: Path) -> Path:
+    return tmp_path / "report.json"
+
+
+class Test_unblob_with_report:
+    def test_output_is_valid_json(
+        self, input_file: Path, extract_root: Path, report_file: Path
+    ):
+        assert not report_file.exists()
+
+        cmd = [
+            sys.executable,
+            "-m",
+            "unblob.cli",
+            "--extract-dir",
+            str(extract_root),
+            "--report",
+            str(report_file),
+            str(input_file),
+        ]
+        print(f"command=\n{' '.join(cmd)}")
+        result = subprocess.run(cmd, executable=sys.executable)
```

Comment:
I didn't expect this test to execute the unblob cli in a subprocess. Can you elaborate? Not even `test_cli` does that.

Reply:
At the time this test was written the logic to write the report was part of the `cli.py`, so that must have been called, however I do not remember the reason anymore, it is definitely no longer valid anymore.